File: Web_DienThoai.client/src/api/flashsale.ts
// src/api/flashsale.ts
import apiClient from './client';

export interface FlashSaleItem {
  productId: number;
  productName: string;
  productSlug: string;
mainImageUrl: string;
  originalPrice: number;
  salePrice: number;
  stockLimit?: number;
  averageRating?: number;
  reviewCount: number;
  soldCount: number;
}

export interface FlashSale {
  id: number;
  name: string;
  startAt: string;
  endAt: string;
  items: FlashSaleItem[];
}

export const flashSaleApi = {
  /** Flash sale đang diễn ra (nếu có). Trả null nếu 404. */
  getActive: () =>
    apiClient
      .get<FlashSale>('/flash-sale/active')
   .then((r) => r.data)
    .catch((err) => {
        if (err?.response?.status === 404) return null;
        throw err;
      }),
};

File: Web_DienThoai.Server/Models/FlashSale.cs
namespace Web_NoiThat.Server.Models;

public class FlashSale
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime StartAt { get; set; }
    public DateTime EndAt { get; set; }
    public bool IsActive { get; set; } = true;
    public ICollection<FlashSaleItem> Items { get; set; } = [];
}

public class FlashSaleItem
{
    public int Id { get; set; }
    public int FlashSaleId { get; set; }
    public FlashSale FlashSale { get; set; } = null!;
    public int ProductId { get; set; }
    public Product Product { get; set; } = null!;
    public decimal SalePrice { get; set; }
    public int? StockLimit { get; set; }    // Số lượng tối đa trong flash sale
    public int SoldQuantity { get; set; }   // Số lượng đã bán trong flash sale
}
